docs/DeveloperGuide.md

```diff
@@ -2,14 +2,219 @@
+
+The processEditTransaction method follows a structured approach to ensure the successful editing of
+transactions while maintaining data integrity:
+
+1. Input Validation: The method begins by validating the user input to ensure it is not empty or null. If
+   the input is invalid, an EmptyArgumentException is thrown to prompt the user to provide valid input.
+2. Transaction Index Extraction: After validating the input, the method extracts the index of the
+   transaction to be edited from the input string. It ensures the extracted data is a valid integer; otherwise, a
+   NumberFormatException is thrown to indicate invalid input.
+3. Index Calculation: Once a valid transaction index is obtained, the method calculates the corresponding
+   index in the transactions ArrayList. As ArrayList indices start from 0, the provided index is decremented
+   by 1 to align with the ArrayList index.
+4. Index Bounds Verification: The method verifies whether the calculated index falls within the bounds of
+   the transactions ArrayList. If the index is out of bounds, an InvalidIndexException is thrown to notify the user of
+   the invalid index provided.
+5. Transaction Editing: Upon successful validation, the method retrieves the transaction object at the
+   calculated index from the transactions ArrayList. It prompts the user to input the updated transaction details
+   through the UserInterface.getEditInformation() method. The edited transaction is then parsed using the
+   parser.parseTransactionType() method to ensure its validity and association with the provided account. Finally, the
+   edited transaction replaces the original transaction at the specified index in the transactions ArrayList using the
+   transactions.set() method.
+6. User Notification: After editing the transaction, the method notifies the user of the successful
+   operation by displaying a message through the UserInterface.printUpdatedTransaction() method.
+
+Sequence Diagram
+The following sequence diagram illustrates the sequence of interactions involved in the editing of a transaction:
+![](./uml/uml-images/processEditTransactionDiagram.png)
+
+### [Implemented] List feature
+
+### Description
+
+The list feature allows users to view their existing transactions. This feature includes viewing all the transactions,
+past week's transactions, past month's transactions and transactions from a specified date range.
+
+#### Implementation
+
+This feature is facilitated through the `TransactionList#processList`, where it prompts the user for their choice of
+list they wish to see. After the user inputs the choice, the get functions will store the transactions in a new
+ArrayList and `UserInterface#printPastTransactions` will handle the output of the list.
+
+Sequence Diagram
+The following sequence diagram illustrates the sequence of interactions involved in the editing of a transaction:
+![](./uml/uml-images/processList.png)
 
 ## Product scope
+
 ### Target user profile
 
```

Comment:
The Target user profile and the Users Stories should be included for to give us a better understanding of this app.